File: app/src/main/java/com/github/willjgriff/ethereumwallet/mvp/MvpPresenter.java
package com.github.willjgriff.ethereumwallet.mvp;

import io.reactivex.disposables.Disposable;

/**
 * Created by Will on 15/08/2016.
 */

public interface MvpPresenter<VIEW> {

	void bindView(VIEW mvpView);

	void unbindView();

	/**
	 * When implementing this method, asynchronous operations should be converted to observables
	 * and their disposables should be added with {@link BaseMvpPresenterKotlin#addDisposable(Disposable)}.
	 * This will ensure callbacks are only executed between {@link BaseMvpPresenterKotlin#bindView(Object)}
	 * and {@link BaseMvpPresenterKotlin#unbindView()}, preventing View access when it isn't available.
	 */
	void viewReady();
}

File: app/src/main/kotlin/com/github/willjgriff/ethereumwallet/mvp/MvpPresenter.kt
package com.github.willjgriff.ethereumwallet.mvp

/**
 * Created by williamgriffiths on 04/05/2017.
 */
interface MvpPresenter<VIEW> {

    fun bindView(mvpView: VIEW)

    fun unbindView()

    /**
     * When implementing this method, asynchronous operations should be converted to observables
     * and their disposables should be added with [BaseMvpPresenterKotlin.addDisposable].
     * This will ensure callbacks are only executed between [BaseMvpPresenterKotlin.bindView]
     * and [BaseMvpPresenterKotlin.unbindView], preventing View access when it isn't available.
     */
    fun viewReady()
}